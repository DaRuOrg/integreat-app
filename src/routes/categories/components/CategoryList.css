@import url('../../../modules/common/constants/colors.css');
@import url('../../../modules/common/constants/helpers.css');

.list a {
  @mixin remove-a;
}
