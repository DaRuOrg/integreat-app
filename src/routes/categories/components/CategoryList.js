import React from 'react'
import PropTypes from 'prop-types'

import CategoryModel from 'modules/endpoint/models/CategoryModel'
import RemoteContent from 'modules/common/components/RemoteContent'

import ContentList from './ContentList'

import style from './CategoryList.css'

/**
 * Displays a ContentList which is a list of categories, a caption and a thumbnail
 */
class CategoryList extends React.Component {
  static propTypes = {
    parentCategory: PropTypes.instanceOf(CategoryModel).isRequired,
    categories: PropTypes.arrayOf(PropTypes.instanceOf(CategoryModel)).isRequired
  }

  render () {
    return (
      <div>
        <div className={style.horizontalLine}>
          <div className={style.heading}>
            <img className={style.headingImage} src={this.props.parentCategory.thumbnail} />
            <div className={style.headingText}>{this.props.parentCategory.title}</div>
            <RemoteContent className={style.shortText}
                           dangerouslySetInnerHTML={{__html: this.props.parentCategory.content}} />
          </div>
        </div>
        <ContentList categories={this.props.categories} />
      </div>
    )
  }
}

export default CategoryList
