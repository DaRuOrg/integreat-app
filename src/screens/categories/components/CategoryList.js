// @flow

import React from 'react'

import Caption from 'modules/common/components/Caption'
import CategoryListItem from './CategoryListItem'
import CategoryModel from '../../../modules/endpoint/models/CategoryModel'

type PropsType = {
  categories: Array<{| model: CategoryModel, subCategories: Array<CategoryModel> |}>,
  title?: string,
  content?: string,
  /** A search query to highlight in the categories titles */
  query?: string
}

/**
 * Displays a ContentList which is a list of categories, a caption and a thumbnail
 */
class CategoryList extends React.Component<PropsType> {
  render () {
    const {categories, title, content, query} = this.props
    return (
      <div>
        {title && <Caption title={title} />}
        <RemoteContent centered dangerouslySetInnerHTML={{__html: content}} />
        <React.Fragment>
          {categories.map(({model, subCategories}) =>
            <CategoryListItem key={model.id}
                              category={model}
                              subCategories={subCategories}
                              query={query} />
          )}
        </React.Fragment>
      </div>
    )
  }
}

export default CategoryList
