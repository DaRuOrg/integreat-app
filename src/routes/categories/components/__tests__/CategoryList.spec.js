import React from 'react'
import { shallow } from 'enzyme'

import CategoryList from '../CategoryList'

const title = 'Willkommen'
const content = 'test content'

const categories = [
  {
    model: {
      id: 35,
      url: '/augsburg/de/willkommen/willkommen-in-augsburg',
      title: 'Willkommen in Augsburg'
    },
    children: [
      {
        id: 35,
        url: '/augsburg/de/willkommen/willkommen-in-augsburg',
        title: 'Willkommen in Augsburg'
      }]
  },
  {
    model: {
      id: 35,
      url: '/augsburg/de/willkommen/willkommen-in-augsburg',
      title: 'Willkommen in Augsburg'
    },
    children: []
  }
]

describe('CategoryList', () => {
  it('should render and display a caption', () => {
    const wrapper = shallow(
      <CategoryList categories={categories}
                    title={title}
                    content={content} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should render and not display a caption', () => {
    const wrapper = shallow(
      <CategoryList categories={categories} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
