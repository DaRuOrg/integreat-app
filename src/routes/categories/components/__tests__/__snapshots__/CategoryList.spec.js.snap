// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList should render and display a caption 1`] = `
<div>
  <Caption
    title="Willkommen"
  />
  <RemoteContent
    centered={true}
    dangerouslySetInnerHTML={
      Object {
        "__html": "test content",
      }
    }
  />
  <div
    className="list"
  >
    <CategoryListItem
      category={
        Object {
          "id": 35,
          "title": "Willkommen in Augsburg",
          "url": "/augsburg/de/willkommen/willkommen-in-augsburg",
        }
      }
      key="35"
    />
    <CategoryListItem
      category={
        Object {
          "id": 35,
          "title": "Willkommen in Augsburg",
          "url": "/augsburg/de/willkommen/willkommen-in-augsburg",
        }
      }
      key="35"
    />
  </div>
</div>
`;

exports[`CategoryList should render and not display a caption 1`] = `
<div>
  <RemoteContent
    centered={true}
    dangerouslySetInnerHTML={
      Object {
        "__html": undefined,
      }
    }
  />
  <div
    className="list"
  >
    <CategoryListItem
      category={
        Object {
          "id": 35,
          "title": "Willkommen in Augsburg",
          "url": "/augsburg/de/willkommen/willkommen-in-augsburg",
        }
      }
      key="35"
    />
    <CategoryListItem
      category={
        Object {
          "id": 35,
          "title": "Willkommen in Augsburg",
          "url": "/augsburg/de/willkommen/willkommen-in-augsburg",
        }
      }
      key="35"
    />
  </div>
</div>
`;
